packages/teams-js/test/public/pages.spec.ts

```diff
@@ -156,96 +163,1296 @@ describe('AppSDK-TeamsAPIs', () => {
+        contentUrl: 'someContentUrl',
+        websiteUrl: 'someWebsiteUrl',
+        entityId: 'someEntityId',
+      };
+
+      Object.values(FrameContexts).forEach(context => {
+        if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
+          it(`pages.getConfig should successfully get settings when initialized with ${context} context`, async () => {
+            await framelessPostMocks.initializeWithContext(context);
+            const promise = pages.getConfig();
+            const message = framelessPostMocks.findMessageByFunc('settings.getSettings');
+            framelessPostMocks.respondToMessage({
+              data: {
+                id: message.id,
+                args: [expectedSettings],
+              },
+            } as DOMMessageEvent);
+
+            await expect(promise).resolves.toEqual(expectedSettings);
+          });
+        } else {
+          it(`pages.getConfig should not allow calls from ${context} context`, async () => {
+            await framelessPostMocks.initializeWithContext(context);
+            expect(() => pages.getConfig()).rejects.toThrowError(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
+    });
+
+    describe('Testing pages.navigateCrossDomain function', () => {
+      const allowedContexts = [
+        FrameContexts.content,
+        FrameContexts.sidePanel,
+        FrameContexts.settings,
+        FrameContexts.remove,
+        FrameContexts.task,
+        FrameContexts.stage,
+        FrameContexts.meetingStage,
+      ];
+
+      it('pages.navigateCrossDomain should not allow calls before initialization', async () => {
+        await expect(pages.navigateCrossDomain('https://valid.origin.com')).rejects.toThrowError(
+          'The library has not yet been initialized',
+        );
+      });
+
+      Object.keys(FrameContexts).forEach(k => {
+        const context = FrameContexts[k];
+        if (allowedContexts.some(allowedContext => allowedContext === context)) {
+          it(`pages.navigateCrossDomain should allow calls from ${context} context`, async () => {
+            expect.assertions(6);
+            await framelessPostMocks.initializeWithContext(context);
+
+            const promise = pages.navigateCrossDomain('https://valid.origin.com');
+            const navigateCrossDomainMessage = framelessPostMocks.findMessageByFunc('navigateCrossDomain');
+            expect(navigateCrossDomainMessage).not.toBeNull();
+            framelessPostMocks.respondToMessage({
+              data: {
+                id: navigateCrossDomainMessage.id,
+                args: ['https://valid.origin.com'],
+              },
+            } as DOMMessageEvent);
+
+            await expect(promise).resolves.not.toThrow();
+            await expect(promise).resolves.toBeUndefined();
```

Comment:
stupid question, how does this work?  wouldn't the promise already be resolved at the second test, also is the second test that useful?